Testing To Do list Part 1

Jord4N-AC opened the pull request:
> ### Colaborators
> @Jord4N-AC 
> @lucas-crodrigues 

In this project, We wrote unit tests for the adding and the deleting functionalities.

### Project requirements

- Use jest framework for testing
- Create a test file ([..].test.js) for a file containing the add item and delete functions that will be tested
- Check if  functions are pure functions, use mocks to manipulate the DOM
- Mock HTML to test if add/delete functions add or remove exactly one <li> element
- Make sure you group your tests using the describe() method

### Optional

- Add a --coverage flag to the test script in package.json, to see the percentage coverage of the application

### General Requirements

- Check linter errors
- Use the correct GitHub flow
- Document my work in a professional way
- Follow best practices for HTML, CSS & JavaScript
- Open a pull request and ask for a review

lucas-crodrigues committed:
Install jest and babel

lucas-crodrigues committed:
Test jest and babel functionalities

lucas-crodrigues committed:
Delete testing files

lucas-crodrigues committed:
Add coverage and jsdom test environment

lucas-crodrigues committed:
Add test for the createAppendTask function

Jord4N-AC committed:
Install jest, jest-enviroment and @babel/plugin-transform-modules-commonjs

Jord4N-AC committed:
Add test for the removeTask function

Jord4N-AC committed:
Add coverage folder to .gitignore

Jord4N-AC committed:
Fix linter errors

Jord4N-AC committed:
Update .gitignore

Jord4N-AC committed:
Disable stylelint in coverage/lcov-report/base.css to fix linter error

Whoistolu commented on src/modules/create_append.test.js:
```diff
@@ -0,0 +1,13 @@
+import createAppendTask from './create_append.js';
+
+test('Add new item to the list', () => {
+  document.body.innerHTML = `
+    <div>
+      <ul id="list-tasks">          
+      </ul>
+    </div>`;
+  const element = document.getElementById('list-tasks');
+  createAppendTask('task1', false, element);
+  const list = document.querySelectorAll('.task-item');
+  expect(list).toHaveLength(1);
+});
```
- [x] [REQUIRED] Great job team, however, we need to group our tests using the the `describe()` method, this breaks your test suite into components. Depending on your test strategy, you might have a describe for each function. Take a look at this example 
```
describe('my beverage', () => {
  test('is delicious', () => {
    expect(myBeverage.delicious).toBeTruthy();
  });

  test('is not sour', () => {
    expect(myBeverage.sour).toBeFalsy();
  });
});
```

Kindly, do this for other tests as well. Happy coding 👍🏽

lucas-crodrigues replied:
Thank you for your feedback, we didn't see this requirement 👍

Whoistolu reviewed (changes requested):
Hi Team
@Jord4N-AC and @lucas-crodrigues 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [x]  _Check the inline comments under the review for sections to improve._
- [x] [REQUIRED] Kindly ensure you have the README file updated by including the details of the second partner in this project and also stating to the user to run `npm test` after running `npm i`. This will make your project more professional. 

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

lucas-crodrigues committed:
Remove unwanted files from the repo

lucas-crodrigues committed:
Move both tests into a single test file

lucas-crodrigues committed:
Fix linters

Jord4N-AC committed:
Update README.md

Including details of the second partner, specify to User to run `npm test` after running `npm i`

GzimAsani reviewed (approved):
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

Jord4N-AC commented:
@GzimAsani
Excuse me could you give more details about 'creating a @current_user', I'd like to know about that

GzimAsani commented:
> @GzimAsani Excuse me could you give more details about 'creating a @current_user', I'd like to know about that

That wasn't meant for you. I apologize for the confusion

Jord4N-AC commented:
> > @GzimAsani Excuse me could you give more details about 'creating a @current_user', I'd like to know about that
> 
> That wasn't meant for you. I apologize for the confusion

@GzimAsani
It's ok no problem
